compiler/test/fail_compilation/fail155.d

```diff
@@ -1,8 +1,7 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/fail155.d(20): Error: overlapping initialization for `y`
-fail_compilation/fail155.d(20):        `struct` initializers that contain anonymous unions must initialize only the first member of a `union`. All subsequent non-overlapping fields are default initialized
+fail_compilation/fail155.d(19): Error: overlapping initialization for field `x` and `y`
```

Comment:
Why not keep the more informative error message?

Reply:
Struct literal expressions now share code with struct initializers, so they get the same error messages. Struct initializers happen to have less supplemental error messages.

Reply:
I added the supplemental error to Struct Initializers as well